scarb/src/compiler/compilers/executable.rs

```diff
@@ -0,0 +1,49 @@
+use crate::compiler::helpers::write_json;
+use crate::compiler::helpers::{build_compiler_config, collect_main_crate_ids};
+use crate::compiler::{CairoCompilationUnit, CompilationUnitAttributes, Compiler};
+use crate::core::{TargetKind, Workspace};
+use anyhow::{Context, Result};
+use cairo_lang_compiler::db::RootDatabase;
+use cairo_lang_executable::executable::Executable;
+
+pub struct ExecutableCompiler;
+
+impl Compiler for ExecutableCompiler {
+    fn target_kind(&self) -> TargetKind {
+        TargetKind::EXECUTABLE.clone()
+    }
+
+    fn compile(
+        &self,
+        unit: CairoCompilationUnit,
+        db: &mut RootDatabase,
+        ws: &Workspace<'_>,
+    ) -> Result<()> {
+        let target_dir = unit.target_dir(ws);
+        let main_crate_ids = collect_main_crate_ids(&unit, db);
+        let compiler_config = build_compiler_config(db, &unit, &main_crate_ids, ws);
+        let executable = Executable::new(
+            cairo_lang_executable::compile::compile_executable_in_prepared_db(
+                db,
+                None,
+                main_crate_ids,
+                compiler_config.diagnostics_reporter,
+            )
+            .with_context(|| {
+                format!(
+                    "Failed to compile executable {}",
+                    unit.main_component().target_name()
+                )
+            })?,
```

Comment:
```suggestion
            )?,
```
Generally, it's the right thing to do / good instinct, but it's not really needed in this context.